src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -34,16 +36,23 @@ public OrderController(ShoppingCartService shoppingCartService,
     }
 
     @PostMapping("/complete")
-    public OrderResponseDto completeOrder(@RequestParam Long userId) {
-        ShoppingCart cart = shoppingCartService.getByUser(userService.get(userId));
+    public OrderResponseDto completeOrder(Authentication authentication) {
+        ShoppingCart cart = shoppingCartService
+                .getByUser(getUserByPrincipal(authentication));
         return orderResponseDtoMapper.mapToDto(orderService.completeOrder(cart));
     }
 
     @GetMapping
-    public List<OrderResponseDto> getOrderHistory(@RequestParam Long userId) {
-        return orderService.getOrdersHistory(userService.get(userId))
+    public List<OrderResponseDto> getOrderHistory(Authentication authentication) {
+        return orderService.getOrdersHistory(getUserByPrincipal(authentication))
                 .stream()
                 .map(orderResponseDtoMapper::mapToDto)
                 .collect(Collectors.toList());
     }
+
+    private User getUserByPrincipal(Authentication authentication) {
+        String login = authentication.getName();
+        return userService.findByEmail(login).orElseThrow(() ->
+                new EntityNotFoundException("Can't find user with login: " + login));
+    }
```

Comment:
You create this method twice (here and in ShoppingCartController) but its body is the same. Maybe you can move it to some service class and use it everywhere you need?

Reply:
good idea, ty for advice